lib/pure/collections/deques.nim

```diff
@@ -50,59 +49,78 @@ runnableExamples:
 
 import std/private/since
 
-import math
+import math, hashes
 
 type
   Deque*[T] = object
     ## A double-ended queue backed with a ringed `seq` buffer.
     ##
-    ## To initialize an empty deque,
-    ## use the `initDeque proc <#initDeque,int>`_.
+    ## To initialize an empty deque with a given capacity use
+    ## `initDeque proc <#initDeque,int>`_.
     data: seq[T]
     head, tail, count, mask: int
 
 const
-  defaultInitialSize* = 4
+  nimDequeDefaultInitialCapacity* {.intdefine.} = 4
 
-template initImpl(result: typed, initialSize: int) =
-  let correctSize = nextPowerOfTwo(initialSize)
-  result.mask = correctSize - 1
-  newSeq(result.data, correctSize)
+template initImpl(result: typed, initialCapacity: int) =
+  assert isPowerOfTwo(initialCapacity)
+  result.mask = initialCapacity-1
+  newSeq(result.data, initialCapacity)
 
 template checkIfInitialized(deq: typed) =
-  when compiles(defaultInitialSize):
+  when declared(nimDequeDefaultInitialCapacity):
     if deq.mask == 0:
-      initImpl(deq, defaultInitialSize)
+      initImpl(deq, nimDequeDefaultInitialCapacity)
 
-proc initDeque*[T](initialSize: int = defaultInitialSize): Deque[T] =
-  ## Creates a new empty deque.
+proc initDeque*[T](initialCapacity: int = nimDequeDefaultInitialCapacity): Deque[T] =
+  ## Create a new empty deque with a given capacity. An implicitly defined
+  ## deque will have a capacity of 0 and be grown to fit elements on the first
+  ## data insertion.
   ##
-  ## Optionally, the initial capacity can be reserved via `initialSize`
-  ## as a performance optimization
-  ## (default: `defaultInitialSize <#defaultInitialSize>`_).
-  ## The length of a newly created deque will still be 0.
+  ## Optionally, the initial capacity can be reserved via `initialCapacity`
+  ## as a performance optimization. The length of a newly created deque will
+  ## still be 0.
+  ##
+  ## ``initialCapacity`` must be a power of two (default: 4).
```

Comment:
```suggestion
  ## `initialCapacity` must be a power of two (default: 4).
```

> In nim sources, prefer single backticks to double backticks since it's simpler and nim doc supports it (even in rst files with nim rst2html).
https://nim-lang.github.io/Nim/contributing.html